lib/Controller/DaemonConfigController.php

```diff
@@ -59,11 +59,37 @@ public function registerDaemonConfig(array $daemonConfigParams, bool $defaultDae
 	#[PasswordConfirmationRequired]
 	public function updateDaemonConfig(string $name, array $daemonConfigParams): Response {
 		$daemonConfig = $this->daemonConfigService->getDaemonConfigByName($name);
+
+		// Safely check if "haproxy_password" exists before accessing it
+		$haproxyPassword = $daemonConfigParams['deployConfig']['haproxy_password'] ?? null;
+
+		// Restore the original password if "dummySecret123" is provided
+		if ($haproxyPassword === 'dummySecret123') {
+			$daemonConfigParams['deployConfig']['haproxy_password'] = $daemonConfig->getDeployConfig()['haproxy_password'] ?? "";
```

Comment:
`deploy_config` instead of `deployConfig`

Reply:
Keen eye, thanks for your attention!